win32ss/gdi/ntgdi/font.c

```diff
@@ -464,19 +463,19 @@ NtGdiAddFontResourceW(
     if ((cwc <= 1) || (cwc > UNICODE_STRING_MAX_CHARS))
         return 0;
 
-    SafeFileName.MaximumLength = (USHORT)(cwc * sizeof(WCHAR));
-    SafeFileName.Length = SafeFileName.MaximumLength - sizeof(UNICODE_NULL);
+    SafeFileName.Length = (USHORT)((cwc - 1) * sizeof(WCHAR));
+    SafeFileName.MaximumLength = SafeFileName.Length + 2 * sizeof(UNICODE_NULL); // Security issue
```

Comment:
Why "security issue" ? If the code is correctly conceived, one single terminating NULL should be enough. Unless the string is actually a multi-string (NUL-terminated strings appended to each other, with a final NUL-terminator, like: `Str1\0Str2\0....StrN\0\0` ).

Reply:
Removed comment.
Made double NUL a single NUL.